modules/storages/app/services/storages/file_link_sync_service.rb

```diff
@@ -26,205 +26,89 @@
 # See COPYRIGHT and LICENSE files for more details.
 #++
 
-# Handle synchronization of FileLinks with external data store such as Storages
 class Storages::FileLinkSyncService
-  FILESINFO_URL_PATH = "/ocs/v1.php/apps/integration_openproject/filesinfo".freeze
+  using Storages::Peripherals::ServiceResultRefinements
 
   def initialize(user:)
     @user = user
-    @service_result = ServiceResult.success(result: [])
   end
 
   def call(file_links)
-    @file_links = file_links
-    storage_file_link_hash = @file_links.group_by(&:storage_id)
-    storage_file_link_hash.each do |storage_id, storage_file_links|
-      # Add new types of storages here. We currently only support Nextcloud
-      sync_nextcloud(storage_id, storage_file_links)
+    file_links
+      .group_by(&:storage_id)
+      .map { |storage_id, storage_file_links| sync_nextcloud(storage_id, storage_file_links) }
+      .reduce(ServiceResult.success(result: [])) do |state, sync_result|
+      state >> ->(state_result) do
+        sync_result.match(
+          on_success: ->(sr) { ServiceResult.success(result: state_result + sr) },
+          on_failure: ->(_) { ServiceResult.success(result: state_result) }
+        )
+      end
     end
-
-    @service_result
   end
 
   private
 
-  # Get the OAuthClientToken that will authenticate us against Nextcloud
-  def get_connection_manager(storage_id)
-    oauth_client = OAuthClient.find_by(integration_id: storage_id)
-    ::OAuthClients::ConnectionManager.new(user: @user, oauth_client:)
+  def sync_nextcloud(storage_id, file_links)
+    ::Storages::Peripherals::StorageRequests
+      .new(storage: ::Storages::Storage.find(storage_id))
+      .files_info_query
+      .call(user: @user, file_ids: file_links.map(&:origin_id))
+      .map(&to_hash)
```

Comment:
🟡 I'd expand this block as this notation might lead people to wonder if it is correct or missing the `:`

Reply:
I could also make `to_hash` an actual function that does not return a function and use `.map(:method(to_hash))`?

wdyt?

Reply:
You could make it into a function that takes a parameter and write the block in a straight forward manner `map {|info| info_to_hash(info) }` would be clearer, IMO.